flutter-for-android.md

```diff
@@ -0,0 +1,1864 @@
+  http.Response response = await http.get(dataURL);
+  setState(() {
+    widgets = JSON.decode(response.body);
+  });
+}
+{% endprettify %}
+
+To update the UI you would call setState which would trigger the build method 
+to run again and update the data.
+
+Here is the full example of loading data asynchronously and displaying it in a 
+ListView:
+
+<!-- skip -->
+{% prettify dart %}
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+
+void main() {
+  runApp(new SampleApp());
+}
+
+class SampleApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Sample App',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new SampleAppPage(),
+    );
+  }
+}
+
+class SampleAppPage extends StatefulWidget {
+  SampleAppPage({Key key}) : super(key: key);
+
+  @override
+  _SampleAppPageState createState() => new _SampleAppPageState();
+}
+
+class _SampleAppPageState extends State<SampleAppPage> {
+  List widgets = new List();
+
+  @override
+  void initState() {
+    super.initState();
+
+    loadData();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+        appBar: new AppBar(
+          title: new Text("Sample App"),
+        ),
+        body: new ListView.builder(
+            itemCount: widgets.length,
+            itemBuilder: (BuildContext context, int position) {
+              return getRow(position);
+            }));
+  }
+
+  Widget getRow(int i) {
+    return new Padding(padding: new EdgeInsets.all(10.0), child: new Text("Row ${widgets[i]["title"]}"));
+  }
+
+  loadData() async {
+    String dataURL = "https://jsonplaceholder.typicode.com/posts";
+    http.Response response = await http.get(dataURL);
+    setState(() {
+      widgets = JSON.decode(response.body);
+    });
+  }
+}
+{% endprettify %}
+
+
+
+## What is the equivalent of an AsyncTask or IntentService on Android
+
+In Android, when you want to access a Network resource you would
+typically spawn an AsyncTask that would run code outside of the UI thread
+to prevent your UI from blocking. The AsyncTask has a thread pool that manages
+the threads for you. 
+
+Since Flutter is single threaded and runs an event loop (like Node.js), you
+don't have to worry about thread management or spawing AsyncTasks or IntentServices.
+
+To run code asynchronously you can declare the function as an async function 
+and await on long running tasks in the function
+
+<!-- skip -->
+{% prettify dart %}
+loadData() async {
+  String dataURL = "https://jsonplaceholder.typicode.com/posts";
+  http.Response response = await http.get(dataURL);
+  setState(() {
+    widgets = JSON.decode(response.body);
+  });
+}
+{% endprettify %}
+
+This is how you would typically do network or database calls. 
+
+With regards to AsyncTask which follows a three step model onPreExecute, 
```

Comment:
this was confusing. I'm not an Android developer, but I'm not sure that you've explained the mapping between AsyncTask and the example above very well. I think think there is a "doInBackground" in Flutter, unless you are talking Isolates. So I don't think AsyncTask is really mappable to Flutter.

Reply:
Fixed